### ISSUE TITLE
Migration of tests to React Native Testing Library

### ISSUE THREAD

**Issue by pranshuchittora** [react-native-elements/react-native-elements#2922]
**Description**
Currently, we are using enzyme & react-test-renderer. Following the standards of the ecosystem, people are shifting from this to **Jext x React Native Testing Library**

Why?
1. It's built specifically for React Native, unlike the test renderer which focuses only on React not Native.
2. It's mature and renowned in the community, therefore it would be easier to mock and test any dependency in future easily (because of the official docs*).

I would love to discuss and hear out your thoughts as well 📢

**Comment by khushal87** [comment]
This would really be great as most of the React native projects uses Jest and React native testing library now, therefore we can work on it. @pranshuchittora. I would like to have a look and work on this issue

**Comment by arpitBhalla** [comment]
@khushal87 may I too work on this please?

**Comment by khushal87** [comment]
> @khushal87 may I too work on this? Please

Sure, let's collaborate, no worries.

**Comment by pranshuchittora** [comment]
@khushal87 @arpitBhalla this is just an RFC, therefore we require some more comments to take the decision, collectively.

**Comment by khushal87** [comment]
Okay sure sir

**Comment by maxikgreat** [comment]
Ye, that's a good solution @pranshuchittora. Moving to ```@testing-library/react-native``` is a cool idea. A couple of projects what I was working on already used it and this library had left mostly a good impression for me. Additionally, it has a similar API to familiar web frameworks and good docs as well 💪 

However, as always, refactoring will take some time

**Comment by flyingcircle** [comment]
^I agree with the move. Maintaining snapshots can be difficult and of dubious value. (People would have to actually review them and understand what their changes mean which usually doesn't happen).

**Comment by maxikgreat** [comment]
> ^I agree with the move. Maintaining snapshots can be difficult and of dubious value. (People would have to actually review them and understand what their changes mean which usually doesn't happen).

Could it be a new idea for GSoC? @flyingcircle Why not?

**Comment by Dhananjayan-PN** [comment]
@pranshuchittora, I definitely feel that moving to Jext x React Native Testing Library would be a great decision. I agree with @flyingcircle, when writing tests after developing a component I find myself looking at the snapshots more than once to understand how the component is laid out, which defeats the purpose of writing unit tests in making sure the tests you write check what you "know" about the component.

**Comment by flyingcircle** [comment]
@maxikgreat Yes I would find this an acceptable thing for gsoc.

**Comment by mhimanshu712** [comment]
I like this idea as well!

**Comment by hsavit1** [comment]
would be great for the testing ecosystem to have demonstrations in an exemplary library

**Comment by arpitBhalla** [comment]
Closes #2922
---
- [ ] Accessory
- [ ] Avatar
- [ ] Badge
- [ ] BottomSheet
- [ ] Button
- [ ] ButtonGroup
- [ ] Card
- [ ] CheckBox
- [ ] Chip
- [ ] Divider
- [ ] FAB
- [ ] FeaturedTile
- [ ] Header
- [ ] Icon
- [ ] Image
- [ ] Input
- [ ] LinearProgress
- [ ] ListItem
- [ ] Overlay
- [ ] PricingCard
- [ ] SearchBar
- [ ] SearchBar-android
- [ ] SearchBar-default
- [ ] SearchBar-ios
- [ ] Slider
- [ ] SocialIcon
- [ ] SpeedDial
- [ ] Switch
- [x] Text
- [x] ThemeProvider
- [ ] Tile
- [ ] Tooltip
- [ ] Tooltip.android
- [ ] colors
- [ ] common
- [ ] makeStyles
- [ ] tab
- [ ] withBadge
- [ ] withTheme

<details open>
<summary>Files to Change</summary>

```diff
./avatar/__tests__/Accessory.tsx
./avatar/__tests__/Avatar.tsx
./badge/__tests__/Badge.tsx
./badge/__tests__/withBadge.tsx
./bottomSheet/__tests__/BottomSheet.tsx
./buttons/__tests__/Button.tsx
./buttons/__tests__/ButtonGroup.tsx
./buttons/__tests__/Chip.tsx
./buttons/__tests__/FAB.tsx
./buttons/__tests__/SpeedDial.tsx
./card/__tests__/Card.tsx
./checkbox/__tests__/CheckBox.tsx
./config/__tests__/ThemeProvider.tsx
./config/__tests__/colors.tsx
./config/__tests__/makeStyles.tsx
./config/__tests__/withTheme.tsx
./divider/__tests__/Divider.tsx
./header/__tests__/Header.tsx
./icons/__tests__/Icon.tsx
./image/__tests__/Image.tsx
./input/__tests__/Input.tsx
./linearProgress/__tests__/LinearProgress.tsx
./list/__tests__/ListItem.tsx
./overlay/__tests__/Overlay.tsx
./pricing/__tests__/PricingCard.tsx
./searchbar/__tests__/SearchBar-android.tsx
./searchbar/__tests__/SearchBar-default.tsx
./searchbar/__tests__/SearchBar-ios.tsx
./searchbar/__tests__/SearchBar.tsx
./searchbar/__tests__/common.tsx
./slider/__tests__/Slider.tsx
./social/__tests__/SocialIcon.tsx
./switch/__tests__/Switch.tsx
./tab/__tests__/tab.tsx
./text/__tests__/Text.tsx
./tile/__tests__/FeaturedTile.tsx
./tile/__tests__/Tile.tsx
./tooltip/__tests__/Tooltip.android.tsx
./tooltip/__tests__/Tooltip.tsx
```
</details>

**Comment by arpitBhalla** [comment]
- [ ] Accessory
- [ ] Avatar
- [ ] Badge
- [ ] BottomSheet
- [ ] Button
- [ ] ButtonGroup
- [ ] Card
- [ ] CheckBox
- [ ] Chip
- [ ] Divider
- [ ] FAB
- [ ] FeaturedTile
- [ ] Header
- [ ] Icon
- [ ] Image
- [ ] Input
- [ ] LinearProgress
- [ ] ListItem
- [ ] Overlay
- [ ] PricingCard
- [ ] SearchBar
- [ ] SearchBar-android
- [ ] SearchBar-default
- [ ] SearchBar-ios
- [ ] Slider
- [ ] SocialIcon
- [ ] SpeedDial
- [ ] Switch
- [x] Text
- [x] ThemeProvider
- [ ] Tile
- [ ] Tooltip
- [ ] Tooltip.android
- [ ] colors
- [ ] common
- [ ] makeStyles
- [ ] tab
- [ ] withBadge
- [ ] withTheme

<details>
<summary>Files to Change</summary>

```diff
./avatar/__tests__/Accessory.tsx
./avatar/__tests__/Avatar.tsx
./badge/__tests__/Badge.tsx
./badge/__tests__/withBadge.tsx
./bottomSheet/__tests__/BottomSheet.tsx
./buttons/__tests__/Button.tsx
./buttons/__tests__/ButtonGroup.tsx
./buttons/__tests__/Chip.tsx
./buttons/__tests__/FAB.tsx
./buttons/__tests__/SpeedDial.tsx
./card/__tests__/Card.tsx
./checkbox/__tests__/CheckBox.tsx
./config/__tests__/ThemeProvider.tsx
./config/__tests__/colors.tsx
./config/__tests__/makeStyles.tsx
./config/__tests__/withTheme.tsx
./divider/__tests__/Divider.tsx
./header/__tests__/Header.tsx
./icons/__tests__/Icon.tsx
./image/__tests__/Image.tsx
./input/__tests__/Input.tsx
./linearProgress/__tests__/LinearProgress.tsx
./list/__tests__/ListItem.tsx
./overlay/__tests__/Overlay.tsx
./pricing/__tests__/PricingCard.tsx
./searchbar/__tests__/SearchBar-android.tsx
./searchbar/__tests__/SearchBar-default.tsx
./searchbar/__tests__/SearchBar-ios.tsx
./searchbar/__tests__/SearchBar.tsx
./searchbar/__tests__/common.tsx
./slider/__tests__/Slider.tsx
./social/__tests__/SocialIcon.tsx
./switch/__tests__/Switch.tsx
./tab/__tests__/tab.tsx
./text/__tests__/Text.tsx
./tile/__tests__/FeaturedTile.tsx
./tile/__tests__/Tile.tsx
./tooltip/__tests__/Tooltip.android.tsx
./tooltip/__tests__/Tooltip.tsx
```
</details>
